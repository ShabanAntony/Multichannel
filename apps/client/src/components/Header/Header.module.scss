.base {
}

.links {
  display: flex;
  flex-direction: row;
  justify-content: center;
  gap: 1rem;
  margin: 2rem 0;

  &_item {
    font-size: 1.5rem;
    text-decoration: none;
    cursor: pointer;
    opacity: 0.8;

    &__selected {
      font-weight: 700;
      opacity: 1;
    }
  }
}

.header {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  justify-content: space-between;
  margin: 0 0 1.5rem;

  @media (min-width: 924px) {
    gap: 0;
    flex-direction: row;
  }
}
