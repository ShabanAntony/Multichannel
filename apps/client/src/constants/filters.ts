export enum ItemFilters {
  DISCOUNT_50_60,
  DISCOUNT_60_PLUS,
}
