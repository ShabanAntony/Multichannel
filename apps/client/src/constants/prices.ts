export const laryToUSD = 2.7
export const VSProfit = 0.5
export const nikeProfit = 0.3
export const reebokProfit = 0.3
export const customsTax = 0.18 + 0.02
