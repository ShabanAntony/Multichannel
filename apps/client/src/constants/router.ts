export enum Pages {
  ITEMS = '/multistream/items/:brandType',
}

export enum BrandTypes {
  Classic = 'vs',
  PINK = 'pink',
  NIKE = 'nike',
  REEBOK = 'reebok',
}
