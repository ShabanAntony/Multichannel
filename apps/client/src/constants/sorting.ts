export enum ItemSorting {
  BY_MAX_DISCOUNT,
  BY_MIN_PRICE,
}
