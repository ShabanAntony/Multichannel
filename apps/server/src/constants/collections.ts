export const collectionsToCheckout = [
  // all sale
  'a202a327-de9d-40ac-8dc3-743ce96b870c',
  // panties
  'f5a391f6-2ca9-4ac4-ba2c-64caa6338185',
  // lingerie
  '7508da5f-a50e-4541-ab89-6741846b3606',
  // bras
  'b60bdcec-c132-48c6-8760-49fbd1446674',
  // sport
  '297255c3-5614-4009-ad7d-557378e8818f',
  // Beauty
  '7eaeaf81-f067-430e-80eb-78a5e871d479',
  // Bags & Accessories
  'cd0dd393-bffe-45e2-a899-6e2d243ceff8',
  // brand we love
  '41d80710-9f25-4405-a309-c2ae3388dc09',
]
