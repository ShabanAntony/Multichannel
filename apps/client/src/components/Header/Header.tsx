import React, { FC, PropsWithChildren, memo } from 'react'
import { Link, generatePath, useLocation } from 'react-router-dom'

import classNames from 'classnames'

import { BrandTypes, Pages } from '../../constants/router'

import styles from './Header.module.scss'

export interface Props {
  totalItems: number
  filteredItems: number
  newItems: number
}

export const URLS = [
  {
    title: "Victoria's secret",
    url: generatePath(Pages.ITEMS, { brandType: BrandTypes.Classic }),
  },
  {
    title: 'PINK',
    url: generatePath(Pages.ITEMS, { brandType: BrandTypes.PINK }),
  },
  {
    title: 'Nike',
    url: generatePath(Pages.ITEMS, { brandType: BrandTypes.NIKE }),
  },
  {
    title: 'Reebok',
    url: generatePath(Pages.ITEMS, { brandType: BrandTypes.REEBOK }),
  },
]

export const Header: FC<PropsWithChildren<Props>> = memo(function Header({
  totalItems,
  filteredItems,
  newItems,
  children,
}) {
  const location = useLocation()

  return (
    <div className={styles.base}>
      <div className={styles.links}>
        {URLS.map(({ title, url }) => (
          <Link
            to={url}
            key={title}
          >
            <span
              className={classNames(styles.links_item, {
                [styles.links_item__selected]: url === location.pathname,
              })}
            >
              {title}
            </span>
          </Link>
        ))}
      </div>
      <div className={styles.header}>
        <span>
          Всего: <strong>{totalItems}</strong>, подходящих:{' '}
          <strong>{filteredItems}</strong>, новых: <strong>{newItems}</strong>
        </span>
        {children}
      </div>
    </div>
  )
})
