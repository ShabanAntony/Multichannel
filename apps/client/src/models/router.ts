import { Location } from 'react-router-dom'

import { createEvent, createStore } from 'effector'

import { BrandTypes } from '../constants/router'

export type Params = Partial<{
  brandType: BrandTypes
}>

export const setParams = createEvent<Params>('set url params')
export const $params = createStore<Params>({}).on(
  setParams,
  (_, params) => params,
)

export const setLocation = createEvent<Location>('set url params')
export const $location = createStore<Location>({
  hash: '',
  key: '',
  pathname: '',
  search: '',
  state: undefined,
}).on(setLocation, (_, params) => params)

// $location.watch((location) => {
//   console.log(`>> location: ${JSON.stringify(location, null, 4)}`)
// })
