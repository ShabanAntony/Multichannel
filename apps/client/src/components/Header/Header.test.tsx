import React from 'react'
import { MemoryRouter, generatePath } from 'react-router-dom'
import renderer from 'react-test-renderer'

import { BrandTypes, Pages } from '../../constants/router'
import { Header } from './Header'

describe('Header', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should render correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter
          initialEntries={[
            generatePath(Pages.ITEMS, { brandType: BrandTypes.PINK }),
          ]}
        >
          <Header
            newItems={10}
            filteredItems={10}
            totalItems={100}
          >
            Hello
          </Header>
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
