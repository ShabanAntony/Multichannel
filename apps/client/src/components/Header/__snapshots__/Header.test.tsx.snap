// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="base"
>
  <div
    className="links"
  >
    <a
      href="/multistream/items/vs"
      onClick={[Function]}
    >
      <span
        className="links_item"
      >
        Victoria's secret
      </span>
    </a>
    <a
      href="/multistream/items/pink"
      onClick={[Function]}
    >
      <span
        className="links_item links_item__selected"
      >
        PINK
      </span>
    </a>
    <a
      href="/multistream/items/nike"
      onClick={[Function]}
    >
      <span
        className="links_item"
      >
        Nike
      </span>
    </a>
    <a
      href="/multistream/items/reebok"
      onClick={[Function]}
    >
      <span
        className="links_item"
      >
        Reebok
      </span>
    </a>
  </div>
  <div
    className="header"
  >
    <span>
      Всего:
      <strong>
        100
      </strong>
      , подходящих:

      <strong>
        10
      </strong>
      , новых:
      <strong>
        10
      </strong>
    </span>
    Hello
  </div>
</div>
`;
